### PR TITLE
add leaflet.contextmenu w/ npm auto-update

### LINKED ISSUE #7827: [Request] Add Leaflet.contextmenu
**Library name:** leaflet.contextmenu
**Git repository url:** https://github.com/aratcliffe/Leaflet.contextmenu
**npm package url(optional):** 
**License(s):**  MIT licence
**Official homepage:**
**Wanna say something? Leave message here:**

=====================
Notes from cdnjs maintainer:
You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by kennynaoh**
PR for #7827 
@x09326 please help me review this pull request, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/aratcliffe/Leaflet.contextmenu
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/leaflet-contextmenu
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 112
   - Count of watchers: 13
   - Count of forks: 35
 * NPM download stats (optional):
   - Downloads in the last day: 3
   - Downloads in the last week: 107
   - Downloads in the last month: 806

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

npm auto-update config added, close #7827, cc @aratcliffe

### CODE REVIEW

File: ajax/libs/leaflet-contextmenu/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "leaflet-contextmenu",
+  "filename": "leaflet.contextmenu.min.js",
```
Comment (x09326) [inline ajax/libs/leaflet-contextmenu/package.json]: The **leaflet.contextmenu.js** has been minified from **leaflet.contextmenu-src.js**
So I think you can just minify **leaflet.contextmenu.css**
And change the ```"filename"``` to ```"leaflet.contextmenu.js"```

@PeterDaveHello please help me review my opinion.
I'm not sure whether I'm right.

Comment (PeterDaveHello) [inline ajax/libs/leaflet-contextmenu/package.json]: Looks correct.

**Comment by kennynaoh** [comment]
Thank you both.
@x09326 
I have updated the files and fixed the ```package.json```.
Please help me review this PR again.
Thanks.

**Comment by x09326** [comment]
@PeterDaveHello 
I think this PR is okay, please help me review it, thanks.

**Comment by PeterDaveHello** [comment]
Same situation with https://github.com/cdnjs/cdnjs/pull/7857#issuecomment-218092867, would you please confirm with its author that is it possible to change the naming so that we can automatically update the lib on cdnjs?

**Comment by kennynaoh** [comment]
Sure, thanks

**Comment by kennynaoh** [comment]
@x09326 
I have updated the version to v1.1.0 and the author provided the minified css file.
Please help me review it again.
Thanks!

**Comment by PeterDaveHello** [comment]
I'm a little bit confused, there are `leaflet.contextmenu.min.js`, `leaflet.contextmenu-src.js` and `leaflet.contextmenu.js`, two versions of source file? Or `leaflet.contextmenu-src.js` is out-dated and should be remove? https://github.com/aratcliffe/Leaflet.contextmenu/tree/master/dist

**Comment by aratcliffe** [comment]
leaflet.contextmenu-src.js is out of date and I've removed it.

**Comment by PeterDaveHello** [comment]
@aratcliffe Thanks!

**Comment by x09326** [comment]
@aratcliffe 
Could you also remove them on npm?
Because this PR is using npm auto-update.
Thanks!

**Comment by PeterDaveHello** [comment]
@x09326 I guess we can not touch the old versions, may just manually remove them from the PR.

**Comment by x09326** [comment]
@kennynaoh 
Please manually remove ```leaflet.contextmenu-src.js```.

**Comment by kennynaoh** [comment]
@x09326 
I have removed `leaflet.contextmenu-src.js` manually.
Please help me review it again.
Thanks!

**Comment by aratcliffe** [comment]
Do you still need me to remove from npm? If so can you describe the steps I need to take to do so.

**Comment by PeterDaveHello** [comment]
> Do you still need me to remove from npm? If so can you describe the steps I need to take to do so.
@aratcliffe no, we can just remove it manually, thanks for your help.

**Comment by PeterDaveHello** [comment]
@kennynaoh and the old version please.

**Comment by kennynaoh** [comment]
@x09326 
I have add old version, too.
Please help me review it again.
Thanks!

**Comment by x09326** [comment]
@PeterDaveHello 
I think this PR is okay, please help me review it, thanks.

**Comment by PeterDaveHello** [comment]
This branch need to rebase with the laster master branch first.

**Comment by kennynaoh** [comment]
@x09326 
I have rebased to the latest master branch.
Please help me review it again.
Thanks!

**Comment by x09326** [comment]
@PeterDaveHello 
I think this PR is okay, please help me review it, thanks.

**Comment by PeterDaveHello** [comment]
Thank you all.

**Pull request merged**